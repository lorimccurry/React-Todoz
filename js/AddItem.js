import React from 'react'

class AddItem extends React.Component {
  constructor () {
    super()

    this.state = {
      itemInput: ''
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange (e) {
    this.setState({itemInput: e.target.value})
  }

  handleSubmit (e) {
    e.preventDefault()
    this.props.onSubmit(this.state.itemInput)
    this.setState({itemInput: ''})
  }

  render () {
    return (
      <form onSubmit={this.handleSubmit}>
        <fieldset>
          <legend>Gimme tha todoz:</legend>
          <input
            type='text'
            placeholder='Get coffee'
            value={this.state.itemInput}
            onChange={this.handleChange}
            autoFocus
          />
        </fieldset>
      </form>
    )
  }
}

const { string, func } = React.PropTypes
AddItem.propTypes = {
  itemInput: string,
  onSubmit: func
}

export default AddItem
