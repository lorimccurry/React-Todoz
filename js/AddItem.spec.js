import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { spy } from 'sinon'
import AddItem from './AddItem'

describe('AddItem', () => {
  it('AddItem snapshot test', () => {
    const component = shallow(<AddItem />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('should render a form and an input', () => {
    const component = shallow(<AddItem />) 
    expect(component.find('form').length).toEqual(1)
    expect(component.find('input').length).toEqual(1)
  })

  it('should accept input', () => {
    const component = mount(<AddItem />)
    const input = component.find('input')
    input.simulate('change', {target: {value: 'get beer'}})
    expect(component.state('itemInput')).toEqual('get beer')
    expect(input.prop('value')).toEqual('get beer')
  })

  it('should call handleSubmit when form is submitted', () => {
    const addItemSpy = spy()
    const component = mount(<AddItem onSubmit={addItemSpy}/>)
    component.setState({itemInput: 'get beer'})
    const form = component.find('form')
    form.simulate('submit')

    expect(addItemSpy.calledOnce).toEqual(true)
    expect(addItemSpy.calledWith('get beer')).toEqual(true)
  })
})

