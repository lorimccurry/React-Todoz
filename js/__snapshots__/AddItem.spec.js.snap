// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItem AddItem snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Gimme tha todoz:
    </legend>
    <input
      autoFocus={true}
      onChange={[Function]}
      placeholder="Get coffee"
      type="text"
      value=""
    />
  </fieldset>
</form>
`;
